tasks/install.yml

```diff
@@ -93,28 +93,38 @@
     - "{{ vault_log_path }}"
     - "{{ vault_run_path }}"
 
-- name: Ensure TLS directory has been created
-  file:
-    dest: "{{ vault_tls_config_path }}"
-    state: directory
-    owner: "{{ vault_user }}"
-    group: "{{ vault_group }}"
-  when: vault_tls_disable == "false"
+- block:
+  - name: Ensure TLS directory has been created
+    file:
+      dest: "{{ vault_tls_config_path }}"
+      state: directory
+      owner: "{{ vault_user }}"
+      group: "{{ vault_group }}"
+
+  - name: Ensure Vault SSL certificate and key are on the server
+    copy:
+      src: "{{ item.src }}"
+      dest: "{{ item.dest }}"
+      owner: "{{ vault_user }}"
+      group: "{{ vault_group }}"
+      mode: "{{ item.mode }}"
+    with_items:
+      - src: "{{ vault_tls_src_files }}/{{ vault_tls_ca_file }}"
+        dest: "{{ vault_tls_config_path }}/{{ vault_tls_ca_file }}"
+        mode: "0644"
+      - src: "{{ vault_tls_src_files }}/{{ vault_tls_ca_file }}"
+        dest: "{{ vault_tls_cert_key_store }}/{{ vault_tls_ca_file }}"
+        mode: "0644"
+      - src: "{{ vault_tls_src_files }}/{{ vault_tls_cert_file }}"
+        dest: "{{ vault_tls_config_path }}/{{ vault_tls_cert_file }}"
+        mode: "0644"
+      - src: "{{ vault_tls_src_files }}/{{ vault_tls_key_file }}"
+        dest: "{{ vault_tls_config_path }}/{{ vault_tls_key_file }}"
+        mode: "0600"
```

Comment:
Block à éxécuter uniquement si vault_tls_disable == "false"

Reply:
J'ai déjà un block qui comprends les tasks suivantes si et seulement si vault_tls_disable == "false":

[BLOCK]
   1/ Ensure TLS directory has been created
   2/ Ensure Vault SSL certificate and key are on the server
   3/ Add TLS CA certificate to OS keystore
[when: vault_tls_disable == "false"]

Du coup je ne comprends pas ta remarque ?

Reply:
ah je pense savoir pourquoi tu dis mettre un block alors qu'il existe. Le "when" n'a pas été touché du coup n'apparait pas dans les changes afficher dans la Pull Request